assert/assertions_test.go

```diff
@@ -175,173 +125,6 @@ type S4 struct {
-		{S4{[]*Nested{{1, 2}, {3, 4}}}, S4{[]*Nested{{1, "a"}, {3, "b"}}}, true},
-		{S4{[]*Nested{{1, 2}, {3, 4}}}, S4{[]*Nested{{1, "a"}, {2, "b"}}}, false},
-
-		{Nested{&intValue, 2}, Nested{&intValue, 2}, true},
-		{Nested{&Nested{1, 2}, 3}, Nested{&Nested{1, "b"}, 3}, true},
-		{Nested{&Nested{1, 2}, 3}, Nested{nil, 3}, false},
-
-		{
-			Nested{map[interface{}]*Nested{nil: nil}, 2},
-			Nested{map[interface{}]*Nested{nil: nil}, 2},
-			true,
-		},
-		{
-			Nested{map[interface{}]*Nested{"a": nil}, 2},
-			Nested{map[interface{}]*Nested{"a": nil}, 2},
-			true,
-		},
-		{
-			Nested{map[interface{}]*Nested{"a": nil}, 2},
-			Nested{map[interface{}]*Nested{"a": {1, 2}}, 2},
-			false,
-		},
-		{
-			Nested{map[interface{}]Nested{"a": {1, 2}, "b": {3, 4}}, 2},
-			Nested{map[interface{}]Nested{"a": {1, 5}, "b": {3, 7}}, 2},
-			true,
-		},
-		{
-			Nested{map[interface{}]Nested{"a": {1, 2}, "b": {3, 4}}, 2},
-			Nested{map[interface{}]Nested{"a": {2, 2}, "b": {3, 4}}, 2},
-			false,
-		},
-	}
-
-	for _, c := range cases {
-		t.Run(fmt.Sprintf("ObjectsExportedFieldsAreEqual(%#v, %#v)", c.expected, c.actual), func(t *testing.T) {
-			res := ObjectsExportedFieldsAreEqual(c.expected, c.actual)
-
-			if res != c.result {
-				t.Errorf("ObjectsExportedFieldsAreEqual(%#v, %#v) should return %#v", c.expected, c.actual, c.result)
-			}
-
-		})
-	}
-}
-
-func TestCopyExportedFields(t *testing.T) {
-	intValue := 1
-
-	cases := []struct {
-		input    interface{}
-		expected interface{}
-	}{
-		{
-			input:    Nested{"a", "b"},
-			expected: Nested{"a", nil},
-		},
-		{
-			input:    Nested{&intValue, 2},
-			expected: Nested{&intValue, nil},
-		},
-		{
-			input:    Nested{nil, 3},
-			expected: Nested{nil, nil},
-		},
-		{
-			input:    S{1, Nested{2, 3}, 4, Nested{5, 6}},
-			expected: S{1, Nested{2, nil}, nil, Nested{}},
-		},
-		{
-			input:    S3{},
-			expected: S3{},
-		},
-		{
-			input:    S3{&Nested{1, 2}, &Nested{3, 4}},
-			expected: S3{&Nested{1, nil}, &Nested{3, nil}},
-		},
-		{
-			input:    S3{Exported1: &Nested{"a", "b"}},
-			expected: S3{Exported1: &Nested{"a", nil}},
-		},
-		{
-			input: S4{[]*Nested{
-				nil,
-				{1, 2},
-			}},
-			expected: S4{[]*Nested{
-				nil,
-				{1, nil},
-			}},
-		},
-		{
-			input: S4{[]*Nested{
-				{1, 2}},
-			},
-			expected: S4{[]*Nested{
-				{1, nil}},
-			},
-		},
-		{
-			input: S4{[]*Nested{
-				{1, 2},
-				{3, 4},
-			}},
-			expected: S4{[]*Nested{
-				{1, nil},
-				{3, nil},
-			}},
-		},
-		{
-			input:    S5{Exported: Nested{"a", "b"}},
-			expected: S5{Exported: Nested{"a", nil}},
-		},
-		{
-			input:    S6{"a", "b"},
-			expected: S6{"a", ""},
-		},
-	}
-
-	for _, c := range cases {
-		t.Run("", func(t *testing.T) {
-			output := copyExportedFields(c.input)
-			if !ObjectsAreEqualValues(c.expected, output) {
-				t.Errorf("%#v, %#v should be equal", c.expected, output)
-			}
-		})
-	}
-}
-
 func TestEqualExportedValues(t *testing.T) {
```

Comment:
This test should also move to `internal/check`.

Reply:
That tests an assertion, are you sure?

Reply:
OK, never mind.